### PR TITLE
Add dynamic return type extension for Context->getAspect

### PR DESCRIPTION
**Pull request by MK-42**
Like in the last PR, I'm quite new to phpstan in general, and this project specifically.

Don't know if we can use the actual implementation of the method instead of duplicating the switch/case statement here.

### CODE REVIEW

**Comment by sascha-egerer** [comment]
> Don't know if we can use the actual implementation of the method instead of duplicating the switch/case statement here.

I think it's fine  - don't know if there is a better way.

Could you please remove the 4 use statements?
```
FILE: ...sascha-egerer/phpstan-typo3/src/Type/ContextDynamicReturnTypeExtension.php
--------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
--------------------------------------------------------------------------------
  5 | ERROR | [x] Type
    |       |     _HumbugBox069fea7e7fc7\phpDocumentor\Reflection\Types\String_
    |       |     is not used in this file.
    |       |     (SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse)
  7 | ERROR | [x] Type PhpParser\Node\Name is not used in this file.
    |       |     (SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse)
 14 | ERROR | [x] Type PHPStan\Type\TypeCombinator is not used in this
    |       |     file.
    |       |     (SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse)
 22 | ERROR | [x] Type TYPO3\CMS\Extbase\Validation\ValidatorResolver is not
    |       |     used in this file.
    |       |     (SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse)
```

**Comment by MK-42** [comment]
Now only the php 7.1 versions are failing, I think because of typo3-8.7 which is pulled due to the version requirements; This does not yet have the Context-API. How to move forward here?

File: src/Type/ContextDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use TYPO3\CMS\Core\Context\Context;
+use TYPO3\CMS\Core\Context\DateTimeAspect;
+use TYPO3\CMS\Core\Context\LanguageAspect;
+use TYPO3\CMS\Core\Context\TypoScriptAspect;
+use TYPO3\CMS\Core\Context\UserAspect;
+use TYPO3\CMS\Core\Context\VisibilityAspect;
+use TYPO3\CMS\Core\Context\WorkspaceAspect;
```
Comment (sascha-egerer) [inline src/Type/ContextDynamicReturnTypeExtension.php]:
```suggestion
```

Does it help for TYPO3 8.7 if we remove the use statements (add FQN below) and then check in the isMethodSupported if the interface `TYPO3\CMS\Core\Context\AspectInterface` does exist ?

Comment (MK-42) [reply src/Type/ContextDynamicReturnTypeExtension.php]: Using the ::class constants still broke it.
I now used the static strings, this makes the pipeline pass

**Comment by MK-42** [comment]
@sascha-egerer can I do some more about this one?

**Comment by sascha-egerer** [comment]
@mk-42 I‘ll try to take care today

File: src/Type/ContextDynamicReturnTypeExtension.php (hunk 1 above)
Comment (sascha-egerer) [reply src/Type/ContextDynamicReturnTypeExtension.php]: I can't reproduce this. Using `::class` should work fine if you use the FQN and do not `use` it. 
See https://3v4l.org/hTgfM

**Comment by MK-42** [comment]
Thanks for taking care!

**Pull request merged**